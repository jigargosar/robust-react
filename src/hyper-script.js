import h from 'react-hyperscript'
import hh from 'hyperscript-helpers'

export const {div} = hh(h)


export {h}
