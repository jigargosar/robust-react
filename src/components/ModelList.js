import {List, ListItemText, MenuItem} from 'material-ui'
import {map, propOr} from 'ramda'
import {h} from '../hyper-script'

const ModelListItem = ({
  model,
  primaryRenderer = propOr('<primaryRenderer or name not found>', 'name'),
  onClick = () => () => {},
}) =>
  h(MenuItem, {onClick: onClick(model)}, [
    h(ListItemText, {primary: primaryRenderer(model)}),
  ])

const ModelList = ({models, ...other}) =>
  h(List, map(model => h(ModelListItem, {model, ...other}), models))

export {ModelList}
