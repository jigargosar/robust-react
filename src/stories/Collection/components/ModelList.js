// @flow

import {List, ListItemText, MenuItem} from 'material-ui'
import {h} from '../../../hyper-script'

type Model = {name: string}

export type ModelListItemProps = {
  model: Model,
}

const ModelListItem = (props: ModelListItemProps) => {
  return h(MenuItem, [h(ListItemText, props.model.name)])
}

export type ModelListProps = {
  models: Array<Model>,
}

const ModelList = ({models}: ModelListProps) => {
  return h(
    List,
    models.map((model, idx) => h(ModelListItem, {key: idx, model})),
  )
}

export {ModelList, ModelListItem}
