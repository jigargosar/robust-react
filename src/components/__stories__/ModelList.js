import {action} from '@storybook/addon-actions'
import {storiesOf} from '@storybook/react'
import {centerDecorator} from '../../storybook-helpers'
import {h} from '../../hyper-script-utils'
import {createFakeModels} from '../__fixtures__/Model.fixture'
import {ModelList} from '../ModelList'

const story = storiesOf('Components|Model.List', module).addDecorator(
  centerDecorator,
)

story.add('basic', () =>
  h(ModelList, {
    models: createFakeModels(),
    onClick: model => event => action('onClick')(model, event.type),
  }),
)
story.add('with primaryRenderer', () =>
  h(ModelList, {
    models: createFakeModels(),
    primaryRenderer: model => `${model.name} (id:${model.id})`,
    onClick: model => event => action('onClick')(model, event.type),
  }),
)
