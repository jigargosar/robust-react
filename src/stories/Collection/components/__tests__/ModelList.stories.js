import Chance from 'chance'
// import 'dom-testing-library/extend-expect'
import {Paper} from 'material-ui'
import * as R from 'ramda'
import {render} from 'react-testing-library'
import {
  describe,
  expect,
  it,
  specs,
  storiesOf,
} from '../../../../../.storybook/facade'
import {div, h} from '../../../../hyper-script'
import type {Model, ModelListItemProps, ModelListProps} from '../ModelList'
import {ModelList, ModelListItem} from '../ModelList'

const createFakeModels = (): Array<Model> => {
  const chance = Chance(11)
  return R.times(
    (idx: number): Model => ({id: idx, name: chance.country({full: true})}),
    3,
  )
}

const models: Array<Model> = createFakeModels()

storiesOf('Model', module)
  .addDecorator(story =>
    div(
      {
        style: {
          padding: '16px 0',
          backgroundColor: '#ddd',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
        },
      },
      [div({style: {flex: 1, maxWidth: 300}}, [h(Paper, [story()])])],
    ),
  )
  .add('ListItem', () => {
    const props: ModelListItemProps = {model: {id: '0', name: 'Foo bar'}}
    const story = h(ModelListItem, props)
    specs(() =>
      describe('ListItem', () =>
        it('should render without crashing', () => {
          const {container} = render(story)
          expect(container.firstChild).toMatchSnapshot()
        })),
    )
    return story
  })
  .add('List', () => {
    const props: ModelListProps = {models}
    const story = h(ModelList, props)
    specs(() =>
      describe('List', () => {
        it('should render without crashing', () => {
          const {container} = render(story)
          expect(container.firstChild).toMatchSnapshot()
        })
      }),
    )
    return story
  })

// describe()
