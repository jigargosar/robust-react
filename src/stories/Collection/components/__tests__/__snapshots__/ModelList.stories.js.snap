// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model List should match snapshot 1`] = `
<ul
  class="MuiList-root-53 MuiList-padding-54"
>
  <li
    class="MuiButtonBase-root-13 MuiListItem-root-3 MuiListItem-default-6 MuiListItem-gutters-10 MuiListItem-button-11 MuiMenuItem-root-1"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root-16"
    >
      <h3
        class="MuiTypography-root-22 MuiTypography-subheading-29 MuiListItemText-primary-19"
      >
        Ceuta & Melilla
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root-46"
    />
  </li>
  <li
    class="MuiButtonBase-root-13 MuiListItem-root-3 MuiListItem-default-6 MuiListItem-gutters-10 MuiListItem-button-11 MuiMenuItem-root-1"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root-16"
    >
      <h3
        class="MuiTypography-root-22 MuiTypography-subheading-29 MuiListItemText-primary-19"
      >
        Bahamas
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root-46"
    />
  </li>
  <li
    class="MuiButtonBase-root-13 MuiListItem-root-3 MuiListItem-default-6 MuiListItem-gutters-10 MuiListItem-button-11 MuiMenuItem-root-1"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root-16"
    >
      <h3
        class="MuiTypography-root-22 MuiTypography-subheading-29 MuiListItemText-primary-19"
      >
        American Samoa
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root-46"
    />
  </li>
</ul>
`;

exports[`Model ListItem should match snapshot 1`] = `
<li
  class="MuiButtonBase-root-13 MuiListItem-root-3 MuiListItem-default-6 MuiListItem-gutters-10 MuiListItem-button-11 MuiMenuItem-root-1"
  role="menuitem"
  tabindex="-1"
>
  <div
    class="MuiListItemText-root-16"
  >
    <h3
      class="MuiTypography-root-22 MuiTypography-subheading-29 MuiListItemText-primary-19"
    >
      Foo bar
    </h3>
  </div>
  <span
    class="MuiTouchRipple-root-46"
  />
</li>
`;
